src/index.coffee

```diff
@@ -42,12 +42,19 @@ class ConnectApp
     @app.use connect.directory(if typeof @root == "object" then @root[0] else @root)
 
     if @https
-      @server = (http2 || https).createServer
-        key: @https.key || fs.readFileSync __dirname + '/certs/server.key'
-        cert: @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
-        ca: @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
-        passphrase: @https.passphrase || 'gulp'
-        @app
+
+      # first make sure it is always an object
+      if !(typeof (@https) is "object")
+        @https = {}
+
+      # use some defaults when not set. do not touch any other https options
+      # see https://github.com/AveVlad/gulp-connect/issues/172
+      @https.key        = @https.key || fs.readFileSync __dirname + '/certs/server.key'
+      @https.cert       = @https.cert || fs.readFileSync __dirname + '/certs/server.crt'
+      @https.ca         = @https.ca || fs.readFileSync __dirname + '/certs/ca.crt'
+      @https.passphrase = @https.passphrase || 'gulp'
```

Comment:
you have a good point. grabbed your first suggestion, extended it a bit. see my second commit. all tests pass.